File: javascript-closure-angular/API/Client/RequestTimeFilterDepartureSearch.js
goog.provide('API.Client.RequestTimeFilterDepartureSearch');

/**
 * @record
 */
API.Client.RequestTimeFilterDepartureSearch = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.id;

/**
 * @type {!string}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.departureLocationId;

/**
 * @type {!Array<!string>}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.arrivalLocationIds;

/**
 * @type {!API.Client.RequestTransportation}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.transportation;

/**
 * @type {!number}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.travelTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.departureTime;

/**
 * @type {!Array<!API.Client.RequestTimeFilterProperty>}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.properties;

/**
 * @type {!API.Client.RequestRangeFull}
 * @export
 */
API.Client.RequestTimeFilterDepartureSearch.prototype.range;

File: scalaz/src/main/scala/org/openapitools/client/api/RequestTimeFilterDepartureSearch.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import RequestTimeFilterDepartureSearch._

case class RequestTimeFilterDepartureSearch (
  id: String,
departureLocationId: String,
arrivalLocationIds: List[String],
transportation: RequestTransportation,
travelTime: Integer,
departureTime: OffsetDateTime,
properties: List[RequestTimeFilterProperty],
range: Option[RequestRangeFull])

object RequestTimeFilterDepartureSearch {
  import DateTimeCodecs._

  implicit val RequestTimeFilterDepartureSearchCodecJson: CodecJson[RequestTimeFilterDepartureSearch] = CodecJson.derive[RequestTimeFilterDepartureSearch]
  implicit val RequestTimeFilterDepartureSearchDecoder: EntityDecoder[RequestTimeFilterDepartureSearch] = jsonOf[RequestTimeFilterDepartureSearch]
  implicit val RequestTimeFilterDepartureSearchEncoder: EntityEncoder[RequestTimeFilterDepartureSearch] = jsonEncoderOf[RequestTimeFilterDepartureSearch]
}
